core.d.ts

```diff
@@ -435,3 +438,88 @@ console.log(await fileTypeFromBlob(blob));
 ```
 */
 export declare function fileTypeFromBlob(blob: Blob): Promise<FileTypeResult | undefined>;
+
+/**
+Function that allows specifying custom detection mechanisms.
+
+An iterable of detectors can be provided via the fileTypeOptions argument for the {@link FileTypeParser.constructor}.
+
+The detectors are called before the default detections in the provided order.
+
+Custom detectors can be used to add new FileTypeResults or to modify return behaviour of existing FileTypeResult detections.
+
+If the detector returns `undefined`, there are 2 possible scenarios:
+
+	1. The detector has not read from the tokenizer, it will be proceeded with the next available detector.
+	2. The detector has read from the tokenizer (`tokenizer.position` has been increased).
+		 In that case no further detectors will be executed and the final conclusion is that file-type returns undefined.
+		 Note that this an exceptional scenario, as the detector takes the opportunity from any other detector to determine the file type.
+
+Example detector array which can be extended and provided via the fileTypeOptions argument:
+
+```js
+import {FileTypeParser} from 'file-type';
+
+const customDetectors = [
+	async tokenizer => {
+		const unicornHeader = [85, 78, 73, 67, 79, 82, 78]; // "UNICORN" as decimal string
+		const buffer = Buffer.alloc(7);
+		await tokenizer.peekBuffer(buffer, {length: unicornHeader.length, mayBeLess: true});
+		if (unicornHeader.every((value, index) => value === buffer[index])) {
+			return {ext: 'unicorn', mime: 'application/unicorn'};
+		}
+
+		return undefined;
+	},
+];
+
+const buffer = Buffer.from("UNICORN");
+const parser = new FileTypeParser({customDetectors});
+const fileType = await parser.fromBuffer(buffer);
+console.log(fileType);
+```
+
+@param tokenizer - An [`ITokenizer`](https://github.com/Borewit/strtok3#tokenizer) usable as source of the examined file.
```

Comment:
There is no need to mention the type as it's already in the type signature.

Reply:
I wanted explictly mention the fact that this is the way to parse [File](https://developer.mozilla.org/en-US/docs/Web/API/File), as users keep being surprised the Node.js specific function `fileTypeFromFile` cannot be used in a browser environment.